.github/workflows/ca-tests.yml

```diff
@@ -1295,3 +1295,272 @@ jobs:
           name: ca-existing-certs-${{ matrix.os }}
           path: |
             /tmp/artifacts/pki
+
+  # https://github.com/dogtagpki/pki/wiki/Issuing-User-Certificate-with-CMC-Shared-Token
+  cmc-shared-token-test:
+    name: Testing CMC shared token
+    needs: [init, build]
+    runs-on: ubuntu-latest
+    env:
+      SHARED: /tmp/workdir/pki
+    strategy:
+      matrix: ${{ fromJSON(needs.init.outputs.matrix) }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Retrieve runner image
+        uses: actions/cache@v3
+        with:
+          key: pki-ca-runner-${{ matrix.os }}-${{ github.run_id }}
+          path: pki-ca-runner.tar
+
+      - name: Load runner image
+        run: docker load --input pki-ca-runner.tar
+
+      - name: Create network
+        run: docker network create example
+
+      - name: Set up DS container
+        run: |
+          tests/bin/ds-container-create.sh ds
+        env:
+          IMAGE: ${{ needs.init.outputs.db-image }}
+          COPR_REPO: ${{ needs.init.outputs.repo }}
+          HOSTNAME: ds.example.com
+          PASSWORD: Secret.123
+
+      - name: Connect DS container to network
+        run: docker network connect example ds --alias ds.example.com
+
+      - name: Set up PKI container
+        run: |
+          tests/bin/runner-init.sh pki
+        env:
+          HOSTNAME: pki.example.com
+
+      - name: Connect PKI container to network
+        run: docker network connect example pki --alias pki.example.com
+
+      - name: Install CA
+        run: |
+          docker exec pki pkispawn \
+              -f /usr/share/pki/server/examples/installation/ca.cfg \
+              -s CA \
+              -D pki_ds_hostname=ds.example.com \
+              -D pki_ds_ldap_port=3389 \
+              -v
+
+      - name: Install CA admin cert
+        run: |
+          docker exec pki pki-server cert-export ca_signing --cert-file ca_signing.crt
+          docker exec pki pki client-cert-import ca_signing --ca-cert ca_signing.crt
+          docker exec pki pki client-cert-import \
+              --pkcs12 /root/.dogtag/pki-tomcat/ca_admin_cert.p12 \
+              --pkcs12-password Secret.123
+
+      # https://github.com/dogtagpki/pki/wiki/Creating-Issuance-Protection-Certificate
+      - name: Create issuance protection cert
+        run: |
+          # generate cert request
+          docker exec pki pki \
+              -d /etc/pki/pki-tomcat/alias \
+              -f /etc/pki/pki-tomcat/password.conf \
+              nss-cert-request \
+              --subject "CN=CA Issuance Protection" \
+              --csr ca_issuance_protection.csr
+
```

Comment:
@edewata so my guess is that you are suggesting to use pki instead of older tools such as certutil, PKCS10Client, and CRMFPopClient?

Reply:
Yes, the `pki nss` command is still being developed, but it's supposed to be easier to use/automate and provide more functionalities (e.g. customizable CSR/cert extensions), so for upstream I'd recommend using that command, but for downstream we can continue to use the older tools. In the future once `pki nss` has a full parity with the older tools we can recommend it for downstream too.

https://github.com/dogtagpki/pki/wiki/Generating-Certificate-Request-with-PKI-NSS
https://github.com/dogtagpki/pki/wiki/Issuing-Certificate-with-PKI-NSS